pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java

```diff
@@ -0,0 +1,1287 @@
+   *
+   * Read each segment row by row and generate a corresponding Avro file per segment. When building the
+   * corresponding Avro row from Pinot segment row, three cases are possible for  a column:
+   *
+   * (1) Column is a global dictionary column. In this case, we consult the global dictionary to look up
+   * the original value and get the corresponding mapped derived  value. Store this derived value in avro
+   * record.
+   *
+   * (2) Column is a member of user supplied time/time-derived column set. In this case, store the original
+   * value as is in the avro record.
+   *
+   * (3) Column is neither of the above two -- in this case, we simply generate a random value and insert into
+   * Avro record. We don't have to remember this value ever again as it won't be used during query generation.
+   *
+   * NOTE: Currently the data generation is done immediately after global dictionary is built so data generation
+   * step doesn't load global dictionary into memory. The data structures are already in memory. The query generation
+   * is done separately and so it loads the global dictionaries and column name mapping.
+   */
+
+  public void generateAvroFiles()
+      throws Exception {
+    _timeToGenerateAvroFiles.start();
+
+    int totalRows = 0;
+    for (int file = 0 ; file < _numFilesToGenerate; file++) {
+      String pathToSegment = _segmentDir + "/" + _segmentDirectories[file];
+      File segmentIndexDir = new File(pathToSegment);
+      try (PinotSegmentRecordReader recordReader = new PinotSegmentRecordReader(segmentIndexDir, _pinotSchema, null);
+          DataFileWriter<GenericData.Record> avroRecordWriter = new DataFileWriter<>(new GenericDatumWriter<GenericData.Record>(_avroSchema))) {
+        String pathToAvroFile = _outputDir + "/" + _filePrefix + file;
+        LOGGER.info("Using segment {} to generate Avro file {}", pathToSegment, pathToAvroFile);
+        File outputAvroFile = new File(pathToAvroFile);
+        avroRecordWriter.create(_avroSchema, outputAvroFile);
+        GenericRow row = new GenericRow();
+        int rows = 0;
+        while (recordReader.hasNext()) {
+          // read actual row from segment
+          row = recordReader.next(row);
+          buildAvroRow(avroRecordWriter, row);
+          rows++;
+        }
+        totalRows += rows;
+        LOGGER.info("Generated Avro File {} with {} rows", pathToAvroFile, rows);
+      }
+    }
+
+    _timeToGenerateAvroFiles.stop();
+    LOGGER.info("Finished generating {} rows across {} avro files. Time taken {}secs",
+        totalRows, _numFilesToGenerate, _timeToGenerateAvroFiles.elapsed(TimeUnit.SECONDS));
+  }
+
+  private void buildAvroRow(
+      DataFileWriter<GenericData.Record> avroRecordWriter,
+      GenericRow pinotRow) throws Exception {
+    GenericData.Record record = new GenericData.Record(_avroSchema);
+    for (String columnName : pinotRow.getFieldNames()) {
+      Object origValue = pinotRow.getValue(columnName);
```

Comment:
getValue() is deprecated.

Reply:
I don't see it deprecated:

`/**
   * Returns the value for the given field.
   * <p>Before setting the {@code defaultNullValue} for a field by calling {@link #putDefaultNullValue(String, Object)},
   * the value for the field can be {@code null}.
   */
  public Object getValue(String fieldName) {
    return _fieldToValueMap.get(fieldName);
  }`

Reply:
Sorry, my bad. getFieldNames() in line 676 is deprecated.

Reply:
Done